src/mono/mono/utils/mono-threads-api.h

```diff
@@ -137,6 +138,24 @@ For further explanation of what can and can't be done in GC unsafe mode:
 		MONO_STACKDATA (__gc_safe_unbalanced_dummy); \
 		mono_threads_enter_gc_safe_region_unbalanced_internal (&__gc_safe_unbalanced_dummy)
 
+#else /* DISABLE_THREADS */
+
+#define MONO_ENTER_GC_UNSAFE	do {
```

Comment:
is it actually safe to always do this? i was under the impression gc safe/unsafe would have consequences beyond whether other threads can kick off GC.

Reply:
Gc safe/ unsafe only makes sense if there is suspension happening. If there's only a single thread, it's not suspending anything.

Reply:
my understanding would be that managed pointers at any given point could be living in registers (wasm locals) and not visible to the GC until they're flushed to the stack/heap at a safepoint

Reply:
And I mean this is happening today (because as I mentioned the function calls underlying the enter/leave macros are all exiting early without doing anything. For single threaded wasm, this pR just gets rid of the last bits of overhead)

Reply:
OK, sounds fine then, it wasn't clear to me if this was changing behavior. If it's just an optimization that makes it more obvious then 👍

Reply:
> my understanding would be that managed pointers at any given point could be living in registers (wasm locals) and not visible to the GC until they're flushed to the stack/heap at a safepoint

Yea that's true. I'm not sure what does that right now.  I'll keep the PR in draft until I check...

Reply:
It's possible nothing does it right now :-)

Reply:
> It's possible nothing does it right now :-)

Yea that seems to be the case.  If you have a localin the C runtime (not in managed - AOT and interp both put temporaries where they can be scanned) that is the only root for a managed object at a safepoint or a GC, and nothing takes the address of the local, we will miss that root.  For the most part the runtime either uses handles or address-of-local now, but it's not something we've audited.

but nonetheless the enter/leave macros don't do anything relevant.